Unable to install codspeed in self-hosted runners

The codspeed action was not able to install codspeed CLI in our self-hosted runners. The action works fine on github runners.

Error
```
> Run # Validate required inputs
Error: Installation of CodSpeed CLI with version 4.11.0 failed.
Reason: 
Error: Process completed with exit code 1.
```

Failed action: https://github.com/open-telemetry/opentelemetry-go/actions/runs/22638668316/job/65608368522?pr=7998#logs
PR: https://github.com/open-telemetry/opentelemetry-go/pull/7998

@GuillaumeLagrange it is not working. https://github.com/open-telemetry/opentelemetry-go/actions/runs/22678571756/job/65742235545?pr=7998

```
> Run # Validate required inputs
Installer hash verified for version 4.11.0
/home/opc/actions-runner/_work/_temp/af22f54c-66de-4880-a7ee-6cd0f739e5a6.sh: line 105: codspeed: command not found
Error: Process completed with exit code 127.
```

@GuillaumeLagrange, we are now using `Ubuntu 24.04` as the image, but encountered new errors.

- No `make` command is found. Is that normal to make sure `make` is available in a github action?
  
  I use a quick fix here by just executing `sudo apt-get update && sudo apt-get install -y make`

- New error when running codspeed

```
Running the benchmarks
  cd ./bridge/opencensus && go test -run='^$' -bench=. -benchtime=500ms ./...
  warning: flag '-run=^$' is not supported by CodSpeed Go runner, ignoring
  go: downloading go.opencensus.io v0.24.0
  go: downloading github.com/stretchr/testify v1.11.1
  go: downloading github.com/go-logr/logr v1.4.3
  go: downloading github.com/cespare/xxhash/v2 v2.3.0
  go: downloading go.opentelemetry.io/auto/sdk v1.2.1
  go: downloading github.com/go-logr/stdr v1.2.2
  go: downloading github.com/google/uuid v1.6.0
  go: downloading golang.org/x/sys v0.41.0
  go: downloading github.com/davecgh/go-spew v1.1.1
  go: downloading github.com/pmezard/go-difflib v1.0.0
  go: downloading gopkg.in/yaml.v3 v3.0.1
  go: downloading github.com/golang/groupcache v0.0.0-20241129210726-2c02b8208cf8
  WORK=/tmp/go-build367140737
  # testing
  Error: /tmp/.tmpIEK36N/codspeed.go:17:20: undefined: InstrumentHooks
  Error: /tmp/.tmpIEK36N/benchmark.go:142:22: undefined: CurrentTimestamp
  Error: /tmp/.tmpIEK36N/benchmark.go:147:18: undefined: CurrentTimestamp
  Error: /tmp/.tmpIEK36N/benchmark.go:174:22: undefined: CurrentTimestamp
  Error: /tmp/.tmpIEK36N/benchmark.go:464:19: undefined: CurrentTimestamp
  Error: /tmp/.tmpIEK36N/benchmark.go:747:22: undefined: NewInstrumentHooks
  FAIL	go.opentelemetry.io/otel/bridge/opencensus [build failed]
  FAIL	go.opentelemetry.io/otel/bridge/opencensus/internal [build failed]
  FAIL	go.opentelemetry.io/otel/bridge/opencensus/internal/oc2otel [build failed]
  FAIL	go.opentelemetry.io/otel/bridge/opencensus/internal/ocmetric [build failed]
  FAIL	go.opentelemetry.io/otel/bridge/opencensus/internal/otel2oc [build failed]
  FAIL
  [WARN  codspeed_go_runner::runner] Command output: 
  [WARN  codspeed_go_runner::runner] Command error output: 
  Error: Failed to run benchmarks: Failed to run benchmark. Exit status: exit status: 1
  
  Stdout:
  
  
  Stderr:
  
  make: *** [Makefile:188: benchmark/./bridge/opencensus] Error 1
  Error: Error: failed to execute the benchmark process: exit status: 2
  Error: Process completed with exit code 1.
```

https://github.com/open-telemetry/opentelemetry-go/actions/runs/22695372512/job/65800569749?pr=7998

I think you could pull the `docker run --rm -it ubuntu:24.04`, and see if this action is runnable under that image.

--- a/action.yml
+++ b/action.yml
@@ -143,26 +143,13 @@ runs:
           const version = process.env.RUNNER_VERSION
           core.setOutput('hash', hashes[version] || '')
 
-    - shell: bash
-      env:
-        GH_MATRIX: "${{ toJson(matrix) }}"
-        GH_STRATEGY: "${{ toJson(strategy) }}"
-        # This is needed to properly handle quotes and multiline commands in the 'run' input properly, rather than doing `RUN_INPUT_RUN=${{ inputs.run }}` in the script
-        CODSPEED_INPUT_RUN: ${{ inputs.run }}
+    - name: Install CodSpeed runner
+      shell: bash
       run: |
-        # Validate required inputs
-        # (custom message for smoother v4 migration)
-        if [ -z "${{ inputs.mode }}" ]; then
-          echo "::error title=Missing required input 'mode'::The 'mode' input is required as of CodSpeed Action v4. Please explicitly set 'mode' to 'simulation' or 'walltime'. Before, this variable was automatically set to instrumentation on every runner except for CodSpeed macro runners where it was set to walltime by default. See https://codspeed.io/docs/instruments for details."
-          exit 1
-        fi
-
-        # Configure and run codspeed-runner
         RUNNER_VERSION="${{ steps.versions.outputs.runner-version }}"
         VERSION_TYPE="${{ steps.versions.outputs.version-type }}"
 
-        # Install the runner
-        SKIP_HASH_CHECK_WARNING_WARNING="${{ inputs.skip-hash-check-warning }}"
+        SKIP_HASH_CHECK_WARNING="${{ inputs.skip-hash-check-warning }}"
 
         if [ "$VERSION_TYPE" = "latest" ]; then
           if [ "$SKIP_HASH_CHECK_WARNING" != "true" ]; then
@@ -214,6 +201,21 @@ runs:
           echo "Installer hash verified for version $RUNNER_VERSION"
 
           bash "$INSTALLER_TMP" --quiet
+        fi
+
+    - name: Run the benchmarks
+      shell: bash
+      env:
+        GH_MATRIX: "${{ toJson(matrix) }}"
+        GH_STRATEGY: "${{ toJson(strategy) }}"
+        # This is needed to properly handle quotes and multiline commands in the 'run' input properly, rather than doing `RUN_INPUT_RUN=${{ inputs.run }}` in the script
+        CODSPEED_INPUT_RUN: ${{ inputs.run }}
+      run: |
+        # Validate required inputs
+        # (custom message for smoother v4 migration)
+        if [ -z "${{ inputs.mode }}" ]; then
+          echo "::error title=Missing required input 'mode'::The 'mode' input is required as of CodSpeed Action v4. Please explicitly set 'mode' to 'simulation' or 'walltime'. Before, this variable was automatically set to instrumentation on every runner except for CodSpeed macro runners where it was set to walltime by default. See https://codspeed.io/docs/instruments for details."
+          exit 1
         fi
 
         # Build the runner arguments array

